lib/timers/promises.js

```diff
@@ -111,7 +115,70 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  validateObject(options, 'options');
+  const { signal, ref = true, throwOnAbort = true } = options;
+  validateAbortSignal(signal, 'options.signal');
+  validateBoolean(ref, 'options.ref');
+  validateBoolean(throwOnAbort, 'options.throwOnAbort');
+
+  if (signal?.aborted) {
+    if (throwOnAbort) throw new AbortError();
+    return;
+  }
+
+  let onCancel;
+  let notYielded = 0;
+  let passCallback;
+  let abortCallback;
+  const interval = new Timeout(() => {
+    notYielded++;
+    if (passCallback) {
+      passCallback();
+      passCallback = undefined;
+      abortCallback = undefined;
+    }
+  }, after, undefined, true, true);
+  if (!ref) interval.unref();
+  insert(interval, interval._idleTimeout);
+  if (signal) {
+    onCancel = () => {
+      // eslint-disable-next-line no-undef
+      clearInterval(interval);
+      if (abortCallback) {
+        abortCallback(new AbortError());
+        passCallback = undefined;
+        abortCallback = undefined;
+      }
+    };
+    signal.addEventListener('abort', onCancel, { once: true });
+  }
+
+  while (!signal?.aborted) {
+    if (notYielded === 0) {
+      try {
+        await new Promise((resolve, reject) => {
+          passCallback = resolve;
+          abortCallback = reject;
+        });
+      } catch (err) {
+        if (throwOnAbort) {
+          throw err;
+        }
+        return;
+      }
+    }
+    for (; notYielded > 0; notYielded--) {
```

Comment:
Why are we yielding all the missed ticks after abort?

Reply:
As stated in the PR, I wasn't sure if we'd want to yield them or not. I wasn't 100% sure if the generator should yield everything it has already generated or just kill the iteration when the signal was sent. Changing the behaviour would be a simple fix, if you think that's what would expected.